src/lang/base/TypeCache.ml

```diff
@@ -24,22 +24,40 @@ open TypeUtil
 (*                    Library type caching                       *)
 (*****************************************************************)
 
-module StdlibTypeCacher (Q : MakeTEnvFunctor) (R : QualifiedTypes) = struct
+module StdlibTypeCacher
+    (Q : MakeTEnvFunctor)
+    (R : QualifiedTypes)
+    (SR : Rep)
+    (ER : sig
+       type rep
+       val get_loc : rep -> loc
+       val mk_msg_payload_id_address : string -> rep ident
+       val mk_msg_payload_id_uint128 : string -> rep ident
 
-  module MakeTEnv = Q(R)
-  open MakeTEnv
+       (* TODO: This needs to be looked through properly *)
+       val parse_rep : string -> rep
+       val get_rep_str: rep -> string
+     end)
+    (L : sig
```

Comment:
Same as the previous comment, must this be inlined?

Reply:
Fixed.

Reply:
Not quite fixed half an hour ago, but fixed now.